src/sqlfluff/rules/L044.py

```diff
@@ -102,11 +102,24 @@ def _analyze_result_columns(
                         else:
                             # Not an alias. Is it a CTE?
                             if wildcard_table in queries:
-                                # Wildcard refers to a CTE. Analyze it.
+                                # Wildcard refers to a CTE at the same level. Analyze it.
                                 self._analyze_result_columns(
                                     queries.pop(wildcard_table), dialect, queries
                                 )
                             else:
+                                # Is it a nested CTE?
+                                if select_info.cte:
+                                    nested_queries = SelectCrawler.gather_nested_ctes(
+                                        select_info.cte.segments[-1], dialect
+                                    )
+                                    if wildcard_table in nested_queries:
```

Comment:
Note the similarity of these lines to line 104 above. The block above handles CTEs at the same level, while this new block handles nested CTEs.